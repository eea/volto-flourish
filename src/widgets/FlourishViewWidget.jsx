const FlourishViewWidget = (props) => {
  const flourish_url = props.value?.download?.replace(
    '/@@download/flourish_zip',
    '/@@flourish/index.html',
  );
  return props.value ? (
    <div className="flourish-view-widget">
      <iframe
        src={flourish_url}
        height="100%"
        width="100%"
        title="Flourish Widget"
      ></iframe>
    </div>
  ) : null;
};

export default FlourishViewWidget;
