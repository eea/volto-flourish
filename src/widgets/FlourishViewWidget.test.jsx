import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import FlourishViewWidget from './FlourishViewWidget';

const mockStore = configureStore();

const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
});

jest.mock('@eeacms/volto-flourish/components/Flourish', () => () => (
  <div data-testid="flourish-component" />
));

test('renders FlourishViewWidget component', () => {
  const mockValue = {
    download: 'http://example.com/flourish/@@download/flourish_zip',
  };

  const component = renderer.create(
    <Provider store={store}>
      <FlourishViewWidget value={mockValue} />
    </Provider>,
  );

  const json = component.toJSON();

  expect(json).toMatchSnapshot();
});
