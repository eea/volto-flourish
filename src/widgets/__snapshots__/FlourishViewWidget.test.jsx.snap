// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FlourishViewWidget component 1`] = `
<div
  className="flourish-view-widget"
>
  <div
    data-testid="flourish-component"
  />
</div>
`;
